addons/point_of_sale/models/stock_picking.py

```diff
@@ -242,9 +242,50 @@ def _add_mls_related_to_order(self, related_order_lines, are_qties_done=True):
             move.quantity = move.product_uom_qty
         moves_remaining = self - moves_to_assign
         existing_lots = moves_remaining._create_production_lots_for_pos_order(related_order_lines)
+        move_lines_to_create = []
+        mls_qties = []
         if are_qties_done:
             for move in moves_remaining:
-                move.quantity = move.product_uom_qty
+                move.move_line_ids.quantity = 0
+                for line in lines_data[move.product_id.id]['order_lines']:
+                    sum_of_lots = 0
+                    for lot in line.pack_lot_ids.filtered(lambda l: l.lot_name):
+                        if line.product_id.tracking == 'serial':
+                            qty = 1
```

Comment:
is it possible to have a qty different than 1 in this case? I feel like this condition is useless and we could just use: abs(line.qty)

Reply:
If you add 2 serial product it will create a line with quantity = 2. It's this code that will create 2 separate lines for each serial number